src/core/settings.c

```diff
@@ -501,10 +503,8 @@ QuicSettingApply(
         if (MinimumMtu > MaximumMtu) {
             return FALSE;
         }
-        if (Destination->MinimumMtu != MinimumMtu) {
+        if (Source->IsSet.MinimumMtu || Source->IsSet.MaximumMtu) {
             Destination->IsSet.MinimumMtu = TRUE;
-        }
-        if (Destination->MaximumMtu != MaximumMtu) {
```

Comment:
I don't understand this removal

Reply:
Thats to solve the 2nd bug found. Basically if we set the minimum, we have to implictly set the maximum. Otherwise, when the global settings are loaded in, they can result in an out of sync state.

Reply:
If setting the min doesn't modify the max, then I don't quite understand what the problem is here.

Reply:
Generally, we need to decide priority here. Usually the order is parent < registry < app. But if the app only sets the min and the registry needs to set a max, and they don't conflict that should be supported.

Reply:
Well then we need to change loading from registry and global to be able to error.

Reply:
Why? And where is that error going to go? Right now, any errors in there fail silently and have fallback code.

Reply:
I found a way to do it silently, and added more tests.